File: split.js
const fs = require('fs');
let html = fs.readFileSync('index.html', 'utf8');

// 1. Tách file CSS
const styleStart = html.indexOf('<style>');
if (styleStart !== -1) {
    const styleEnd = html.indexOf('</style>', styleStart);
    if (styleEnd !== -1) {
        const css = html.substring(styleStart + 7, styleEnd).trim();
        fs.writeFileSync('styles.css', css);
        html = html.substring(0, styleStart) + '<link rel="stylesheet" href="styles.css">' + html.substring(styleEnd + 8);
        console.log('✅ Đã tách CSS sang file styles.css');
    }
}

// 2. Tách file JS (chỉ lấy script chính)
let scriptStart = html.lastIndexOf('<script>');
while (scriptStart !== -1) {
    const scriptEnd = html.indexOf('</script>', scriptStart);
    if (scriptEnd !== -1) {
        const js = html.substring(scriptStart + 8, scriptEnd);
        // Nhận diện script chính dựa trên từ khóa
        if (js.includes('let apiKey') || js.includes('flatQuestions')) {
            fs.writeFileSync('main.js', js.trim());
            html = html.substring(0, scriptStart) + '<script src="main.js"></script>' + html.substring(scriptEnd + 9);
            console.log('✅ Đã tách JS sang file main.js');
            break;
        }
    }
    scriptStart = html.lastIndexOf('<script>', scriptStart - 1);
}

fs.writeFileSync('index.html', html);
console.log('✅ Đã cập nhật lại file index.html');

File: split.py
import os

with open('index.html', 'r', encoding='utf-8') as f:
    html = f.read()

# 1. Tách CSS
style_start = html.find('<style>')
if style_start != -1:
    style_end = html.find('</style>', style_start)
    if style_end != -1:
        css = html[style_start+7 : style_end].strip()
        with open('styles.css', 'w', encoding='utf-8') as f:
            f.write(css)
        html = html[:style_start] + '<link rel="stylesheet" href="styles.css">' + html[style_end+8:]
        print("✅ Đã tách CSS sang file styles.css")

# 2. Tách JS
script_start = html.rfind('<script>')
while script_start != -1:
    script_end = html.find('</script>', script_start)
    if script_end != -1:
        js = html[script_start+8 : script_end]
        if 'let apiKey' in js or 'flatQuestions' in js:
            with open('main.js', 'w', encoding='utf-8') as f:
                f.write(js.strip())
            html = html[:script_start] + '<script src="main.js"></script>' + html[script_end+9:]
            print("✅ Đã tách JS sang file main.js")
            break
    script_start = html.rfind('<script>', 0, script_start)

with open('index.html', 'w', encoding='utf-8') as f:
    f.write(html)
print("✅ Đã cập nhật lại file index.html")
